client/syntaxes/bitbake.tmLanguage.json

```diff
@@ -1,270 +1,203 @@
+                    "begin": "(\")",
+                    "end": "(\")",
+                    "patterns": [
+                        {
+                            "include": "#variable-reference"
+                        }
+                    ]
+                },
+                {
+                    "name": "string.quoted.single.bb",
+                    "begin": "'",
+                    "end": "'",
+                    "patterns": [
+                        {
+                            "include": "#variable-reference"
+                        }
+                    ]
+                }
+            ]
         },
-        {
-          "include": "#string"
-        }
-      ]
-    },
-    "variables": {
-      "patterns": [
-        {
-          "captures": {
-            "1": {
-              "name": "punctuation.definition.variable.bb"
+        "comment": {
+            "match": "(\\s*)((#)(.*))\\n",
+            "captures": {
+                "1": {
+                    "name": "punctuation.whitespace.comment.leading.bb"
+                },
+                "2": {
+                    "name": "comment.line.bb"
+                },
+                "3": {
+                    "name": "comment.line.number-sign.bb"
+                },
+                "4": {
+                    "name": "comment.line.text.bb"
+                }
             }
-          },
-          "match": "(\\$)([a-zA-Z0-9_:])*\\b",
-          "name": "variable.language.bb"
         },
-        {
-          "captures": {
-            "1": {
-              "name": "punctuation.definition.variable.bb"
+        "functions": {
+            "begin": "\\s*(\\s+[a-zA-Z0-9_-][\\w-]*)?(\\.)?([a-zA-Z0-9_-][\\w-]*)(:append|:prepend)?(?=\\s*\\()",
+            "beginCaptures": {
+                "1": {
+                    "name": "variable.other.names.bb"
+                },
+                "2": {
+                    "name": "keyword.operator.bb"
+                },
+                "3": {
+                    "name": "entity.name.function.python.bb"
+                },
+                "4": {
+                    "name": "keyword.other.bitbake-operator.bb"
+                }
+            },
+            "end": "(?=\\))",
+            "patterns": [
+                {
+                    "include": "#function-params"
+                }  
+            ]
+        },
+        "function-content": {
+            "begin": "(?<=\\(.*\\)\\{)",
+            "end": "(?=\\})",
+            "patterns": [
+                {
+                    "include": "#variable-assignment"
+                },
+                {
+                    "include": "#string"
+                },
+                {
+                    "include": "#variable-reference"
+                }
+            ]
+        },
+        "variable-reference": {
+            "patterns": [
+                {
+                    "match": "(\\$\\{)([a-zA-Z0-9_]*)(\\})",
+                    "name": "variable.language.bb",
```

Comment:
The name on this line and the one on line 163 are redundant. One of them can be removed. If this line gets deleted there must be an additional 3. object in the captures object to highlight the closing bracket.